Pint should define a min version for typing_extensions

We found an [issue](https://gitlab.com/taurus-org/taurus/-/issues/1294#note_1427407491) with pint 0.22 due to an old version of `typing_extensions` installed.

```
____________________ ERROR collecting cli/test/test_alt.py _____________________
ImportError while importing test module '/usr/local/lib/python3.9/dist-packages/taurus/cli/test/test_alt.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/usr/local/lib/python3.9/dist-packages/taurus/cli/test/test_alt.py:25: in <module>
    from taurus.core.util.test.test_plugin import mock_entry_point
/usr/local/lib/python3.9/dist-packages/taurus/core/__init__.py:87: in <module>
    from .taurusattribute import *  # noqa: F403,F401
/usr/local/lib/python3.9/dist-packages/taurus/core/taurusattribute.py:39: in <module>
    from taurus.core.units import Quantity
/usr/local/lib/python3.9/dist-packages/taurus/core/units.py:34: in <module>
    from pint import UnitRegistry, __version__
/usr/local/lib/python3.9/dist-packages/pint/__init__.py:28: in <module>
    from .formatting import formatter, register_unit_format
/usr/local/lib/python3.9/dist-packages/pint/formatting.py:20: in <module>
    from .babel_names import _babel_lengths, _babel_units
/usr/local/lib/python3.9/dist-packages/pint/babel_names.py:11: in <module>
    from .compat import HAS_BABEL
/usr/local/lib/python3.9/dist-packages/pint/compat.py:34: in <module>
    from typing_extensions import Self  # noqa
E   ImportError: cannot import name 'Self' from 'typing_extensions' (/usr/lib/python3/dist-packages/typing_extensions.py)
```

`typing_extensions` 3.7.4 is installed but `typing_extensions.Self` was only added in [4.0.0](https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#added-in-version-400)

`pint` should require at least `typing_extensions >= 4.0.0`. Not sure if even a newer version is needed or not.

--- a/CHANGES
+++ b/CHANGES
@@ -12,7 +12,7 @@ Pint Changelog
 - Add `dim_sort` function to _formatter_helpers.
 - Add `dim_order` and `default_sort_func` properties to FullFormatter.
   (PR #1926, fixes Issue #1841)
-- Minimum version requirement added for typing_extensions>=4.0.0. 
+- Minimum version requirement added for typing_extensions>=4.0.0.
   (PR #1996)
 - Documented packages using pint.
   (PR #1960)

--- a/CHANGES
+++ b/CHANGES
@@ -7,6 +7,8 @@ Pint Changelog
 - Add `dim_sort` function to _formatter_helpers.
 - Add `dim_order` and `default_sort_func` properties to FullFormatter.
   (PR #1926, fixes Issue #1841)
+- Minimum version requirement added for typing_extensions>=4.0.0. 
+  (PR #1996)
 
 
 0.23 (2023-12-08)

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
 appdirs>=1.4.4
-typing_extensions
+typing_extensions>=4.0.0
 flexcache>=0.3
 flexparser>=0.3

